motr/m0crate/crate_index.c

```diff
@@ -383,6 +502,11 @@ static int cr_idx_w_init(struct cr_idx_w *ciw,
 	ciw->nr_keys *= ciw->nr_kv_per_op;
 	ciw->ordered_keys = wit->keys_ordered;
 
+	/* If key size is random, generate it using rand(). */
+	if (ciw->wit->key_size < 0)
+		ciw->wit->key_size = ciw->wit->min_key_size +
+			cr_rand_pos_range(ciw->wit->max_key_size);
```

Comment:
Should this be:

s
/cr_rand_pos_range(ciw->wit->max_key_size)
/cr_rand_pos_range(ciw->wit->max_key_size - ciw->wit->min_key_size)

?
Otherwise the key_size might be greater than max.

Reply:
Fixed.